### DISCUSSION TITLE
ERROR: a role referenced in a mapping is not defined

### DISCUSSION THREAD

**Discussion by alanlubold** [gravitational/teleport#33880] [Q&A] [answered]
When trying to create a trusted cluster, I am getting an error:

arc@6-0-0-0-13:/usr/local/bin2 $ sudo ./tctl --config=/etc/teleport2.yaml create /etc/trusted_cluster2.yaml
**ERROR: a role "arc" referenced in a mapping *:[arc] is not defined**

(the reason for all the 2's is that there are two separate instances of Teleport running on the machine. The other one is using a different server and is working fine.)

There is a role called arc, which has been working well for some time on many trusted clusters. The main server was originally version 11, since step-updated to 14. The various members of the trusted clusters were also step-updated, and during this the "tctl create trusted_cluster.yaml" did not need to be run. This is the first time it has been run on a new trusted cluster and it has failed. 

Here is the trusted_cluster2.yaml file, with the server name Xed out:

kind: trusted_cluster
version: v2
metadata:
  name: main2
spec:
  enabled: true
  role_map:
  - local:
     - arc
    remote: '*'
  token: XXX
  tunnel_addr: XXX:3080
  web_proxy_addr: XXX:3080

Here is the roles.yaml file from the server, which was created some time ago, and was just force recreated successfully:

kind: role
version: v5
metadata:
   name: admin
spec:
   allow:
     logins:
     - ubuntu
     - arc
     - pi
     node_labels:
      '*': '*'

Please let me know if there is any more information I need to supply.

Thanks in advance.

**Comment by Venture200** [comment]
Hello

I was getting same problem as well, the reason is that you haven't created the role in your cluster but in root cluster. Try to create the role where you want to add resources, then in trusted cluster add that role definition

> kind: role
metadata:
  name: <role-name>
spec:
  allow:
    logins:
    - admin
    node_labels:
      application: goo
      environment: prod
      zone: fed0
    app_labels:
      application: goo
      environment: prod
      zone: fed0
    rules:
    - resources:
      - node
      verbs:
      - list
      - read
      - ssh
    - resources:
      - app
      verbs:
      - list
      - read
      - connect
    - resources:
      - event
      verbs:
      - list
      - read
    - resources:
      - session
      verbs:
      - read
      - list
      where: contains(session.participants, user.metadata.name)
  deny:
    logins:
    - guest
  options:
    cert_format: standard
    create_db_user: false
    create_desktop_user: false
    create_host_user: false
    desktop_clipboard: true
    desktop_directory_sharing: true
    enhanced_recording:
    - command
    - network
    forward_agent: false
    idp:
      saml:
        enabled: true
    max_session_ttl: 8h0m0s
    pin_source_ip: false
    port_forwarding: true
    record_session:
      default: best_effort
      desktop: true
    ssh_file_copy: true
version: v7

and then in your truster cluster file add this role 
    
>  - local:
    - <role-name>
    remote: <role-name>

**Comment by webvictim** [comment, marked as answer]
As far as I can see, your role is not called `arc`, it's called `admin`?

**Reply by Venture200** [reply]
Hello, @webvictim my app service appear and disappear time to time, what's the cause ? it's not containerized app

**Reply by webvictim** [reply]
Hey - please make a new post about this as it's a separate issue.

**Comment by alanlubold** [comment]
Gus,

 

Thank you for your response. 

 

I did figure it out in the end, I was confused between the role on the server versus the what I thought was a role on the remote machine. 

 

Thanks again,

 

Alan

**Comment by alanlubold** [comment]
Gus,

 

That wasn’t me, not sure how someone hijacked my post. Maybe they copied me on it by mistake. I’m not VentureBoy. 

 

Alan

**Reply by webvictim** [reply]
I know :)
